src/Agent.Worker/Release/Artifacts/GitHubArtifact.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Agent.Worker.Build;
+using Microsoft.VisualStudio.Services.Agent.Worker.Release.Artifacts.Definition;
+using Microsoft.VisualStudio.Services.ReleaseManagement.WebApi;
+using Microsoft.VisualStudio.Services.ReleaseManagement.WebApi.Contracts;
+
+using Newtonsoft.Json;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Release.Artifacts
+{
+    public class GitHubArtifact : AgentService, IArtifactExtension
+    {
+        public Type ExtensionType => typeof(IArtifactExtension);
+        public AgentArtifactType ArtifactType => AgentArtifactType.GitHub;
+
+        public async Task DownloadAsync(
+            IExecutionContext executionContext,
+            ArtifactDefinition artifactDefinition,
+            string localFolderPath)
+        {
+            ArgUtil.NotNull(executionContext, nameof(executionContext));
+            ArgUtil.NotNull(artifactDefinition, nameof(artifactDefinition));
+            ArgUtil.NotNullOrEmpty(localFolderPath, nameof(localFolderPath));
+
+            var gitHubDetails = artifactDefinition.Details as GitHubArtifactDetails;
+            ArgUtil.NotNull(gitHubDetails, nameof(gitHubDetails));
+
+            executionContext.Output(StringUtil.Loc("RMReceivedGithubArtifactDetails"));
+            ServiceEndpoint endpoint = executionContext.Endpoints.FirstOrDefault((e => string.Equals(e.Name, gitHubDetails.ConnectionName, StringComparison.OrdinalIgnoreCase)));
+            if (endpoint == null)
+            {
+                throw new InvalidOperationException(StringUtil.Loc("RMReceivedGithubArtifactDetails", gitHubDetails.ConnectionName));
```

Comment:
RMReceivedGithubArtifactDetails seems to be inappropriate to use in the error message?